libr/crypto/p/crypto_bech32.c

```diff
@@ -0,0 +1,217 @@
+/* radare - BSD-3-Clause - Copyright (c) 2017, 2021, 2024 - Pieter Wuille, W0nda */
+
+#include <r_lib.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+
+static uint32_t bech32_polymod_step(ut32 pre) {
+	uint8_t b = pre >> 25;
+	return ((pre & 0x1FFFFFF) << 5) ^
+		(-((b >> 0) & 1) & 0x3b6a57b2UL) ^
+		(-((b >> 1) & 1) & 0x26508e6dUL) ^
+		(-((b >> 2) & 1) & 0x1ea119faUL) ^
+		(-((b >> 3) & 1) & 0x3d4233ddUL) ^
+		(-((b >> 4) & 1) & 0x2a1462b3UL);
+}
+
+static uint32_t bech32_final_constant(bech32_encoding enc) {
+	if (enc == BECH32_ENCODING_BECH32) {
+		return 1;
+	}
+	if (enc == BECH32_ENCODING_BECH32M) {
+		return 0x2bc830a3;
+	}
```

Comment:
missingn return statement, i would recommend you to use a precondition and return 1 or 0 by default
```suggestion
		return 1;
	}
```

Reply:
If I follow all your suggested changes, this function will return 1 no matter what the "if" statements return,
is this what's expected ?